opentelemetry/src/metrics/instruments/counter.rs

```diff
@@ -4,7 +4,14 @@ use std::sync::Arc;
 
 use super::SyncInstrument;
 
-/// An instrument that records increasing values.
+/// A monotonic instrument that records increasing values.
+///
+/// Counters are used to measure values that only increase over time, such as the number
+/// of requests received, bytes sent, or errors encountered. Only non-negative values
+/// should be recorded. Negative values violate the monotonic property and will be
+/// dropped by the SDK with a warning.
```

Comment:
while its true that the spec says API should not validate this, and instead it is upto the implementations (i.e SDK) to do this validation, the SDK spec does not explicitly state what should it do when the value is negative. I'd love to get a spec clarification before implementing this.

I did take this route long ago (https://github.com/open-telemetry/opentelemetry-dotnet/issues/2519) but never got the time to clarify if this is the right behavior to do in the sdk.